src/sessions.ts

```diff
@@ -591,18 +589,18 @@ function attemptTransaction<TSchema>(
 
   if (!isPromiseLike(promise)) {
     session.abortTransaction().catch(() => null);
-    throw new MongoInvalidArgumentError(
-      'Function provided to `withTransaction` must return a Promise'
+    return Promise.reject(
```

Comment:
Why'd you revert the `throw`?

Reply:
I was just curious, this change is fine.  It'll be moot once Malik gets to this code anyways